### PR TITLE
practice4-task4_BlockingObjectPool

### PR DESCRIPTION
**Pull request by tyroff**
Task 4. (20 points)

Create simple object pool with support for multithreaded environment. No any extra inheritance, polymorphism or generics needed here, just implementation of simple class.

### CODE REVIEW

File: src/practice4/task4/BlockingObjectPool.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    /**
+     * Gets object from pool or blocks if pool is empty
+     *
+     * @return object from pool
+     */
+    public Object get() {
+        while (pool.size() == 0) {
+            try {
+                wait();
```
Comment (UShalamitski) [inline src/practice4/task4/BlockingObjectPool.java:33]: Run your code

File: src/practice4/task4/BlockingObjectPool.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * Pool that block when it has not any items or it full
+ */
+public class BlockingObjectPool {
+
+    private int capacity = 1;
+    private final BlockingQueue<Object> pool = new ArrayBlockingQueue<>(capacity);
```
Comment (UShalamitski) [inline src/practice4/task4/BlockingObjectPool.java:13]: There is no sense to use your own synchronization logic if you have BlocingQueue. It is already synchronized. Use simple async storage

File: src/practice4/task4/BlockingObjectPool.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    /**
+     * Creates filled pool of passed size
+     *
+     * @param size of pool
+     */
+    public BlockingObjectPool(int size) {
+        if (size > 0) capacity = size;
+        else if (size < 1) throw new NoSuchElementException();
```
[the remark this answers is not in the document]
Comment (tyroff) [reply src/practice4/task4/BlockingObjectPool.java:22]: Made changes

File: src/practice4/task4/BlockingObjectPool.java (hunk 1 above)
Comment (tyroff) [reply src/practice4/task4/BlockingObjectPool.java:33]: I changed "while" to "if".

**Pull request merged**